cpp/oneapi/dal/backend/primitives/sparse_blas/set_csr_data_dpc.cpp

```diff
@@ -0,0 +1,129 @@
+                         const std::int64_t row_count,
+                         const std::int64_t column_count,
+                         dal::sparse_indexing indexing,
+                         dal::array<Float> &data,
+                         dal::array<std::int64_t> &column_indices,
+                         dal::array<std::int64_t> &row_offsets,
+                         const std::vector<sycl::event> &deps) {
+    ONEDAL_ASSERT(data.get_count());
+    ONEDAL_ASSERT(column_indices.get_count());
+    ONEDAL_ASSERT(row_offsets.get_count() == row_count + 1);
+    return oneapi::mkl::sparse::set_csr_data(queue,
+                                             dal::detail::get_impl(handle).handle,
+                                             row_count,
+                                             column_count,
+                                             sparse_indexing_to_mkl(indexing),
+                                             const_cast<std::int64_t *>(row_offsets.get_data()),
+                                             const_cast<std::int64_t *>(column_indices.get_data()),
+                                             const_cast<Float *>(data.get_data()),
+                                             deps);
+}
+
+template <typename Float>
+sycl::event set_csr_data(sycl::queue &queue,
+                         sparse_matrix_handle &handle,
+                         const std::int64_t row_count,
+                         const std::int64_t column_count,
+                         dal::sparse_indexing indexing,
+                         const Float *data,
+                         const std::int64_t *column_indices,
+                         const std::int64_t *row_offsets,
+                         const std::vector<sycl::event> &deps) {
+    ONEDAL_ASSERT(data);
+    ONEDAL_ASSERT(column_indices);
+    ONEDAL_ASSERT(row_offsets);
+    return oneapi::mkl::sparse::set_csr_data(queue,
+                                             dal::detail::get_impl(handle).handle,
+                                             row_count,
+                                             column_count,
+                                             sparse_indexing_to_mkl(indexing),
+                                             const_cast<std::int64_t *>(row_offsets),
+                                             const_cast<std::int64_t *>(column_indices),
+                                             const_cast<Float *>(data),
+                                             deps);
+}
+
+ONEDAL_EXPORT sycl::event set_csr_data(sycl::queue &queue,
+                                       sparse_matrix_handle &handle,
+                                       dal::csr_table &table,
+                                       const std::vector<sycl::event> &deps) {
+    ONEDAL_ASSERT(table.has_data());
+    data_type dtype = table.get_metadata().get_data_type(0);
+    ONEDAL_ASSERT(dtype == data_type::float32 || dtype == data_type::float64);
```

Comment:
This functionality already [exists](https://github.com/oneapi-src/oneDAL/blob/f719aa3d9b23378ab7ab89da30fc99887e121b22/cpp/oneapi/dal/detail/common.hpp#L288)